crates/apub/src/objects/mod.rs

```diff
@@ -54,3 +65,39 @@ pub(crate) async fn append_attachments_to_comment(
 
   Ok(content)
 }
+
+pub(crate) async fn handle_community_moderators(
+  new_mods: &Vec<ObjectId<ApubPerson>>,
+  community: &ApubCommunity,
+  context: &Data<LemmyContext>,
+) -> LemmyResult<()> {
+  let community_id = community.id;
+  let current_moderators =
+    CommunityModeratorView::for_community(&mut context.pool(), community_id).await?;
+  // Remove old mods from database which arent in the moderators collection anymore
+  for mod_user in &current_moderators {
+    let mod_id = ObjectId::from(mod_user.moderator.ap_id.clone());
+    if !new_mods.contains(&mod_id) {
+      let community_moderator_form =
+        CommunityModeratorForm::new(mod_user.community.id, mod_user.moderator.id);
+      CommunityActions::leave(&mut context.pool(), &community_moderator_form).await?;
+    }
+  }
+
+  // Add new mods to database which have been added to moderators collection
+  for mod_id in new_mods {
+    // Ignore errors as mod accounts might be deleted or instances unavailable.
+    let mod_user: Option<ApubPerson> = mod_id.dereference(context).await.ok();
+    if let Some(mod_user) = mod_user {
+      if !current_moderators
+        .iter()
+        .map(|c| c.moderator.ap_id.clone())
+        .any(|x| x == mod_user.ap_id)
```

Comment:
I don't think we need the call to `map` here since we're only using the mapped values in the `any` test function. 

```suggestion
        .any(|c| c.moderator.ap_id == mod_user.ap_id)
```